apache-spark/spark3.0/image-classification/src/main/scala/com/examples/SparkPredictor.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.examples
+
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.param.{Param, ParamMap}
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, Dataset, Encoders}
+
+class SparkPredictor(override val uid: String) extends Transformer {
+
+  def this() = this(Identifiable.randomUID("SparkPredictor"))
+
+  final val inputCol = new Param[String](this, "inputCol", "The input column")
+  final val outputCol = new Param[String](this, "outputCol", "The output column")
+  final val modelUrl = new Param[String](this, "modelUrl", "The model URL")
+
+  def setInputCol(value: String): this.type = set(inputCol, value)
```

Comment:
Why use setter? Can we use builder pattern to create it

Reply:
or Scala can use case class

Reply:
This is a common pattern for class that extends Transformer, see https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala#L162